pkg/binary/envoy/run.go

```diff
@@ -78,6 +78,13 @@ func (r *Runtime) RunPath(path string, args []string) error {
 
 	go waitForExit(cmd, waitCancel) // waits in a goroutine. We may need to kill the process if a signal occurs first.
 
+	// Wait up to 2 seconds to log the admin address
+	for i := 0; i < 10 && sigCtx.Err() == nil && r.GetAdminAddress() == ""; i++ {
+		time.Sleep(200 * time.Millisecond)
+	}
+
+	log.Infof("discovered admin address: %v", r.GetAdminAddress())
```

Comment:
this is key, as people can scrape it even if they could also by looking at the file path. the main issue is that envoy doesn't print the resolved port, rather the input (which could be zero). Even if envoy later changes this, we can't change the past so need to work with this.